tarsier/tag_utils.ts

```diff
@@ -237,6 +237,38 @@ function getAllElementsInAllFrames(): HTMLElement[] {
   return allElements;
 }
 
+function hasLabel(element: HTMLElement): boolean {
+  const tagNames = ["input", "textarea", "select", "button"];
+
+  if (!tagNames.includes(element.tagName.toLowerCase())) {
+    return false;
+  }
+
+  const label = document.querySelector(`label[for='${element.id}']`);
```

Comment:
Should make sure we are escaping any special characters in the id
```suggestion
  const escapedId = CSS.escape(element.id);
  const label = document.querySelector(`label[for="${escapedId}"]`);
```